### PR TITLE
Rewrite gokz-quiet file structure, move quiet options to another category, fix show players menu option being in the wrong order

### PR DESCRIPTION
**Pull request by zer0k-z**
The source file is now split into several files, one for menu options and one file for each feature of the plugin.

This PR is related to #434, allows for later changes to be cleaner, when more features are added in the future.

### CODE REVIEW

File: addons/sourcemod/scripting/gokz-quiet/hideplayers.sp [hunk 1]
```diff
@@ -0,0 +1,305 @@
...
+	}
+	if (init)
+	{
+		flags |= USERMSG_INITMSG;
+	}
+
+	DataPack dp = new DataPack();
+	dp.WriteCell(msg_id);
+	dp.WriteCell(newTotal);
+	dp.WriteCellArray(newClients, newTotal);
```
Comment (Ciallo-Ani) [inline addons/sourcemod/scripting/gokz-quiet/hideplayers.sp:99]: Could you append a fix on here in this pr? I tested it on my server(only 1 player online) and error throws
`[SM] Exception reported: Invalid data pack operation: current position isn't an array!`

**DataPack doesn't support CellArray or FloatArray with size 0**
**You have to jump to the next postition**

File: addons/sourcemod/scripting/gokz-quiet/hideplayers.sp [hunk 2]
```diff
@@ -0,0 +1,305 @@
...
+}
+
+static void RequestFrame_WeaponSound(DataPack dp)
+{
+	dp.Reset();
+
+	UserMsg msg_id = dp.ReadCell();
+	int newTotal = dp.ReadCell();
+	int newClients[MAXPLAYERS];	
+	dp.ReadCellArray(newClients, newTotal);
```
Comment (Ciallo-Ani) [inline addons/sourcemod/scripting/gokz-quiet/hideplayers.sp:119]: Could you append a fix on here in this pr? I tested it on my server(only 1 player online) and error throws
[SM] Exception reported: Invalid data pack operation: current position isn't an array!

**DataPack doesn't support CellArray or FloatArray with size 0**
**You have to jump to the next postition**

Comment (zer0k-z) [reply addons/sourcemod/scripting/gokz-quiet/hideplayers.sp:119]: That's strange. The reloading client should always be included in the array, which makes  its size 1 at minimum...

Comment (Ciallo-Ani) [reply addons/sourcemod/scripting/gokz-quiet/hideplayers.sp:119]: Not a listen server.
It seems that this error throws **only with 1 player and GOTV** on the server
I removed GOTV and went fine, maybe it's just a little probleme:eyes:.

Comment (Ciallo-Ani) [reply addons/sourcemod/scripting/gokz-quiet/hideplayers.sp:119]: i also tested that the Callback param `playersNum` **always equal to 0** if only 1 player on the server

Comment (zer0k-z) [reply addons/sourcemod/scripting/gokz-quiet/hideplayers.sp:119]: if `playersNum` is 0, then that means `newTotal` must be 0 as well, and the plugin will return without ever calling `RequestFrame`, so it shouldn't be throwing an exception, right?

**Comment by zer0k-z** [comment]
Latest commit should fix #436 as well.

**Comment by zealain** [comment]
Are the menu category change and #436 fix the only behavioral changes? It's just kinda hard to compare the code after it was split up so I want to make sure I don't miss anything.

**Comment by zer0k-z** [comment]
Yes, the menu should be the only difference between this PR and the current stable version.

**Pull request merged**